src/CoreShop/Bundle/FrontendBundle/DependencyInjection/CompilerPass/RegisterFrontendControllerPass.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * CoreShop.
+ *
+ * This source file is subject to the GNU General Public License version 3 (GPLv3)
+ * For the full copyright and license information, please view the LICENSE.md and gpl-3.0.txt
+ * files that are distributed with this source code.
+ *
+ * @copyright  Copyright (c) 2015-2019 Dominik Pfaffenbauer (https://www.pfaffenbauer.at)
+ * @license    https://www.coreshop.org/license     GNU General Public License version 3 (GPLv3)
+*/
+
+namespace CoreShop\Bundle\FrontendBundle\DependencyInjection\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+class RegisterFrontendControllerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        $controllers = $container->getParameter('coreshop.frontend.controllers');
+
+        foreach ($controllers as $key => $value) {
+            $controllerKey = sprintf('coreshop.frontend.controller.%s', $key);
+            $controllerClass = $container->getParameter($controllerKey);
+
+            if ($container->hasDefinition($controllerClass)) {
+                $customController = $container->getDefinition($controllerClass);
+
+                $customController->addMethodCall('setContainer', [new Reference('service_container')]);
+                $customController->addMethodCall('setTemplateConfigurator', [new Reference('coreshop.frontend.template_configurator')]);
+
+                $container->setDefinition($controllerKey, $customController)->setPublic(true);
+
+                continue;
+            }
+
+            $controllerDefinition = new Definition($controllerClass);
+            $controllerDefinition->addMethodCall('setContainer', [new Reference('service_container')]);
+            $controllerDefinition->addMethodCall('setTemplateConfigurator', [new Reference('coreshop.frontend.template_configurator')]);
+            $controllerDefinition->setPublic(true);
+
+            switch ($key) {
+                case 'security':
+                    $controllerDefinition->setArguments([
+                        new Reference('security.authentication_utils'),
+                        new Reference('form.factory'),
+                        new Reference('coreshop.context.shopper'),
+                    ]);
+                break;
+
+                case 'payment':
+                    $controllerDefinition->setMethodCalls([
+                        ['setContainer', [new Reference('service_container')]]
```

Comment:
already defined at line 42?

Reply:
line 42 also sets the TemplateConfiguration, but the PaymentController is not a FrontendController